### PR TITLE
Add DownloadPageAPI to the autoupdater to download from HTML web pages

### PR DESCRIPTION
**Pull request by Salamandar**
Tested with Jitsi and this configuration:

```
autoupdate.strategy = "latest_webpage_link"
autoupdate.upstream = "https://download.jitsi.org/stable/"
autoupdate.version_regex = "^jitsi-meet-web_([\\d\\.\\-]*)_all.deb$"
```

### CODE REVIEW

**Comment by orhtej2** [comment]
I'm not saying it's a bad idea, I'm saying it's brittle as is web scraping in general 🤔

**Comment by Salamandar** [comment]
The version_regex is matching against the link "content" (name ?).

**Review by OniriCorpe — APPROVED** [review]
LGTM this handy shit

**Comment by fflorent** [comment]
FWIW, I would probably need that for elasticsearch8.

File: tools/autoupdate_app_sources/rest_api.py [hunk 1]
```diff
@@ -176,3 +178,18 @@ def releases(self) -> list[dict[str, Any]]:
...
+
+
+class DownloadPageAPI:
+    def __init__(self, upstream: str) -> None:
+        self.web_page = upstream
+
+    def get_web_page_links(self) -> dict[str, str]:
+        r = requests.get(self.web_page)
+        r.raise_for_status()
+        soup = BeautifulSoup(r.text, features="lxml")
```
Comment (Psycojoker) [inline tools/autoupdate_app_sources/rest_api.py:190]: lxml and bs4 are missing from requirements.txt

**Review by Psycojoker — DISMISSED** [review]

**Comment by alexAubin** [comment]
So ugh what's the status of this ?

File: tools/autoupdate_app_sources/rest_api.py (hunk 1 above)
Comment (Salamandar) [reply tools/autoupdate_app_sources/rest_api.py:220]: Fixed.

**Pull request merged**